sdk/lib/convert/line_splitter.dart

```diff
@@ -10,12 +10,31 @@ const int _CR = 13;
 
 /// A [StreamTransformer] that splits a [String] into individual lines.
 ///
-/// A line is terminated by either a CR (U+000D), a LF (U+000A), a
-/// CR+LF sequence (DOS line ending),
-/// and a final non-empty line can be ended by the end of the string.
+/// A line is terminated by either:
+/// * a CR, carriage return:`\r, U+000D, 0x0D`
+/// * a LF, line feed:`\n, U+000A, 0x0A` (unix line break)
+/// * a CR+LF sequence (DOS line ending)
+/// * a final non-empty line can be ended by the end of the string
 ///
 /// The returned lines do not contain the line terminators.
-
+///
+/// Example:
+/// ```dart
+/// const LineSplitter ls = LineSplitter();
+/// const String sampleText =
+///     'Dart is: \ran object-oriented \nclass-based \ngarbage-collected'
```

Comment:
What is \ran

Reply:
It's a carriage return and "an" - but I agree it's a little bit unclear in this example, especially for beginners

Maybe do this?

`\r an object-oriented \n class-based \n garbage-collected`

Reply:
Yes. I think it is better to change it to:
```
const String sampleText =
    'Dart is: \r an object-oriented \n class-based \n garbage-collected'
    '\r\n language with C-style syntax \r\n';
```
    
Even though output contains then whitespaces but it is easier to spot \r and \n also on html page. 

Output: 
```
0: Dart is: 
1:  an object-oriented 
2:  class-based 
3:  garbage-collected 
4:  language with C-style syntax
```